parachain/primitives/beacon/src/lib.rs

```diff
@@ -193,3 +195,39 @@ pub struct BeaconBlock {
 	pub state_root: H256,
 	pub body: Body,
 }
+
+impl ExecutionHeader {
```

Comment:
`check_receipt_proof` and `apply_merkle_proof` are copied from the `snowbridge-ethereum` primitive crate to implement the new `ExecutionHeader`, but I reuse everything else.